spec/controllers/media_controller_spec.rb

```diff
@@ -204,6 +204,29 @@
 
         expect(response.headers['Cache-Control']).to eq('max-age=86400, public')
       end
+
+      context 'and the replacement is draft' do
+        before do
+          replacement.update_attribute(:draft, true)
+        end
+
+        it 'serves the asset when requested via something other than the draft-assets host' do
+          request.headers['X-Forwarded-Host'] = "not-#{AssetManager.govuk.draft_assets_host}"
+
+          expect(controller).to receive(:proxy_to_s3_via_nginx).with(asset)
+
+          get :download, params
+        end
+
+        it 'redirects to the replacement when requested via the draft-assets host' do
```

Comment:
Minor: "when requested via the draft-assets host" -> "when requested via the draft-assets host by a signed-in user"